binding.cc

```diff
@@ -1015,6 +1025,86 @@ namespace zmq {
 
 #endif
 
+  NAN_METHOD(Socket::Readv) {
+    Socket* socket = GetSocket(info);
+    if (socket->state_ != STATE_READY)
+      return info.GetReturnValue().Set(Nan::Undefined());
+
+    int events;
+    size_t events_size = sizeof(events);
+
+    int rc = 0;
+    int flags = 0;
+    int64_t more = 1;
+    size_t more_size = sizeof(more);
+    size_t index = 0;
+
+    while (true) {
+      rc = zmq_getsockopt(socket->socket_, ZMQ_EVENTS, &events, &events_size);
+      if (rc != 0) {
```

Comment:
I personally hate how that reads in the C-language family: `if (dofoo()) { }` reads like "if dofoo returns trueish, execute this block". But it's the opposite.

I would prefer writing `if (zmq_getsockopt(...) != 0) {`, but have to admit both this and what you propose occur in this file at the moment.